### ISSUE TITLE
invalid register name and invalid operand for instruction in arch/powerpc/kernel/vdso32/gettimeofday.S

### ISSUE THREAD

**Issue by tpimh** [ClangBuiltLinux/linux#764]
Build of ppc32 kernel with integrated-as enabled failing:
```
arch/powerpc/kernel/vdso32/gettimeofday.S:34:20: error: invalid register name
  .cfi_register lr,%r12
                   ^
arch/powerpc/kernel/vdso32/gettimeofday.S:71:10: error: invalid operand for instruction
 cmpli 0,%r3,0
         ^
arch/powerpc/kernel/vdso32/gettimeofday.S:72:10: error: invalid operand for instruction
 cmpli 1,%r3,1
         ^
arch/powerpc/kernel/vdso32/gettimeofday.S:77:20: error: invalid register name
  .cfi_register lr,%r12
                   ^
arch/powerpc/kernel/vdso32/gettimeofday.S:107:16: error: invalid operand for instruction
        cmpl 0,%r8,%r0
               ^
arch/powerpc/kernel/vdso32/gettimeofday.S:160:10: error: invalid operand for instruction
 cmpli 0,%r4,0
         ^
arch/powerpc/kernel/vdso32/gettimeofday.S:189:20: error: invalid register name
  .cfi_register lr,%r12
                   ^
```

**Comment by nickdesaulniers** [comment]
https://developer.ibm.com/articles/l-ppc/ makes it sound like the `%r` convention is unconventional.
https://godbolt.org/z/Yme6zt

cc @arndb 

for `cmpli`, looks like Clang wasn't expecting a register operand, I suppose.

**Comment by daxtens** [comment]
It looks like some of this might be resolved by b2f663073917babb84f5ba9129451e8d564cba1a ("[PowerPC] Allow a '%' prefix for registers in CFI directives"). I'm building llvm from git now to have a look.

**Comment by daxtens** [comment]
However, it seems a bigger issue might be `clang -integrated-as` not liking e.g. `mulld r3, r3, r4` and only accepting `mulld 3, 3, 4`; ppc linux asm is exclusively written with the `r` prefix. Weirdly my reading of `llvm/lib/Target/PowerPC/AsmParser/PPCAsmParser.cpp` is that it  should recognise the `r` prefix, and should have done since 2013, so maybe I'm misunderstanding how all the pieces fit together...

**Comment by daxtens** [comment]
@cmriedl points out that this could be due to llvm's assembler not supporting `-mregnames`: gnu as also doesn't like the `r` prefix unless you pass that option.

**Comment by daxtens** [comment]
@mpe pointed out that our register names are actually #defined, so llvm is actually rejecting %r0, %r1 etc, which is even odder. However, I can work around this for now by just changing the defines, for example `#define r0 0`.

**Comment by nickdesaulniers** [comment]
Yes, from my godbolt link, it looks like `.cfi_directive` is fixed, and `cmpli` is generally broken (regardless of the `%` or not).  So this is two distinct issues.

**Comment by nickdesaulniers** [comment]
cmpli is #1419 . Let's use this issue to track `.cfi_register`.

**Comment by nemanjai** [comment]
The `.cfi_register` issue was fixed on Nov. 18, 2020 with commit `b2f663073917b`.

**Comment by nickdesaulniers** [comment]
Indeed, I can no longer reproduce with:
```sh
$ ARCH=powerpc make LLVM=1 -j72 pseries_defconfig arch/powerpc/kernel/vdso32/
# or
$ ARCH=powerpc make LLVM=1 -j72 ppc44x_defconfig arch/powerpc/kernel/vdso32/
# or
$ ARCH=powerpc make LLVM=1 -j72 powernv_defconfig arch/powerpc/kernel/vdso32/
```
(working around #1484, #1418 and #672).

> The .cfi_register issue was fixed on Nov. 18, 2020 with commit b2f663073917b.

lol, I reviewed that patch...forgot about this issue. Looks like this shipped in clang-12. Thanks @gwelymernans !
